src/java.security.jgss/share/classes/sun/security/krb5/PrincipalName.java

```diff
@@ -607,29 +613,48 @@ public byte[] asn1Encode() throws Asn1Exception, IOException {
 
 
     /**
-     * Checks if two <code>PrincipalName</code> objects have identical values in their corresponding data fields.
+     * Checks if two <code>PrincipalName</code> objects have identical values
+     * in their corresponding data fields.
+     * <p>
+     * If {@systemProperty jdk.security.krb5.name.case.sensitive} is set to true,
+     * the name comparison is case-sensitive. Otherwise, it's case-insensitive.
+     * <p>
+     * It is used in {@link sun.security.krb5.internal.ccache.FileCredentialsCache}
+     * and {@link sun.security.krb5.internal.ktab.KeyTab} to retrieve ccache
+     * or keytab entry for a principal.
      *
      * @param pname the other <code>PrincipalName</code> object.
      * @return true if two have identical values, otherwise, return false.
      */
-    // It is used in <code>sun.security.krb5.internal.ccache</code> package.
     public boolean match(PrincipalName pname) {
         boolean matched = true;
         //name type is just a hint, no two names can be the same ignoring name type.
         // if (this.nameType != pname.nameType) {
         //      matched = false;
         // }
         if ((this.nameRealm != null) && (pname.nameRealm != null)) {
```

Comment:
isn't 'nameRealm' non-null? Is this check necessary? Having this check seems to imply their value may be null and it seems incomplete that there is no code handling the cases when one or both of them are null.

Reply:
You're right. The `realm` field used to be nullable but it cannot be now. I'll fix it. Thanks.